According to Danby, if we use astronomical units (au) for $a$ and years for $P$, then we can eliminate the denomenator in the relation

$$ \frac{P^2}{P_E^2} = \frac{a^3}{a_E^3} $$

because the values for Earth, $P_E$ and $a_E$, are one au and one year, yielding

$$ P^2 = a^3 $$

We should be able to solve for $P$ and $a$ in order to calculate approximate orbital period and mean distance from the sun, given one of the known values.

In [2]:
import math
from sympy import *
from sympy import init_printing
init_printing(use_latex='mathjax')

P = Symbol("P", positive=True, real=True)
a = Symbol("a", positive=True, real=True)
eq = Eq(P**2, a**3)

solve(eq, a), solve(eq, P)

⎛⎡ 2/3⎤  ⎡ 3/2⎤⎞
⎝⎣P   ⎦, ⎣a   ⎦⎠

Thus

$$ a = P^\frac{2}{3} \quad \textrm{and} \quad P = {a}^\frac{3}{2} $$

The known values for $P$ and $a$ for the major planets of the solar system follow. On Wikipedia the mean distance to the Sun is labeled as the semi-major axis, but I believe the two are equivalent.

| Planet        | Orbital Period ($P$) | Mean Distance from Sun ($a$) |
| ------------- | -------------------- | ---------------------------- |
| Mercury       | 0.240846             | 0.387098                     |
| Verus         | 0.615198             | 0.723332                     |
| Earth         | 1.0                  | 1.0                          |
| Mars          | 1.88085              | 1.523679                     |
| Jupiter       | 11.862               | 5.2044                       |
| Saturn        | 29.4571              | 9.5826                       |
| Uranus        | 84.0205              | 19.2185                      |
| Neptune       | 164.8                | 30.07                        |

In [3]:
orbital_period = lambda a: a**(3/2)
mean_distance = lambda P: P**(2/3)

In [7]:
planets = [
    ("Mercury", 0.240846, 0.387098),
    ("Venus",   0.615198, 0.723332),
    ("Earth",   1.0,      1.0     ),
    ("Mars",    1.88085,  1.523679),
    ("Jupiter", 11.862,   5.2044  ),
    ("Saturn",  29.4571,  9.5826  ),
    ("Uranus",  84.0205,  19.2185 ),
    ("Neptune", 164.8,    30.07   )
]

def error_magnitude(known, value):
    e = abs((known - value) / known);
    if e == 0:
        return -math.inf
    else:
        return math.log10(e)

for planet in planets:
    name, P_known, a_known = planet;
    P = orbital_period(a_known)
    a = mean_distance(P_known)
    P_error = error_magnitude(P_known, P)
    a_error = error_magnitude(a_known, a)
    print(f"| {name:<10} "
          f"| {P:.6f} | {P_known:.6f} | {P_error:.2f}"
          f"| {a:.6f} | {a_known:.6f} | {a_error:.2f}"
          f"|"
         )

| Mercury    | 0.240842 | 0.240846 | -4.73| 0.387103 | 0.387098 | -4.91|
| Venus      | 0.615186 | 0.615198 | -4.71| 0.723341 | 0.723332 | -4.89|
| Earth      | 1.000000 | 1.000000 | -inf| 1.000000 | 1.000000 | -inf|
| Mars       | 1.880790 | 1.880850 | -4.49| 1.523712 | 1.523679 | -4.67|
| Jupiter    | 11.872878 | 11.862000 | -3.04| 5.201221 | 5.204400 | -3.21|
| Saturn     | 29.663681 | 29.457100 | -2.15| 9.538059 | 9.582600 | -2.33|
| Uranus     | 84.251809 | 84.020500 | -2.56| 19.183308 | 19.218500 | -2.74|
| Neptune    | 164.892211 | 164.800000 | -3.25| 30.058788 | 30.070000 | -3.43|


| Planet | Orbital Period ($P$) | Known Value | Error | Mean Distance ($a$) | Known Value | Error |
| ------ | -------------------: | ----------: | ----: | ------------------: | ----------: | ----: |
| Mercury    | 0.240842 | 0.240846 | -4.73| 0.387103 | 0.387098 | -4.91|
| Venus      | 0.615186 | 0.615198 | -4.71| 0.723341 | 0.723332 | -4.89|
| Earth      | 1.000000 | 1.000000 | -$\infty$| 1.000000 | 1.000000 | -$\infty$|
| Mars       | 1.880790 | 1.880850 | -4.49| 1.523712 | 1.523679 | -4.67|
| Jupiter    | 11.872878 | 11.862000 | -3.04| 5.201221 | 5.204400 | -3.21|
| Saturn     | 29.663681 | 29.457100 | -2.15| 9.538059 | 9.582600 | -2.33|
| Uranus     | 84.251809 | 84.020500 | -2.56| 19.183308 | 19.218500 | -2.74|
| Neptune    | 164.892211 | 164.800000 | -3.25| 30.058788 | 30.070000 | -3.43|

